silk/middleware.py

```diff
@@ -142,28 +143,29 @@ def process_request(self, request):
         request_model = RequestModelFactory(request).construct_request_model()
         DataCollector().configure(request_model, should_profile=should_profile)
 
-    @transaction.atomic()
```

Comment:
Looking at the `transaction.atomic` implementation in https://github.com/albertyw/django/blob/master/django/db/transaction.py#L316, this looks like it can be modified into `transaction.atomic(using=router.db_for_write(models.SQLQuery))` instead of converting into a `with` block

Other examples: https://sourcegraph.com/search?q=context%3Aglobal+%40transaction.atomic%28using&patternType=keyword&sm=0&df=%5B%22lang%22%2C%22Python%22%2C%22lang%3Apython%22%5D&__cc=1